### PR TITLE
Make Bolas great again

### PR DESCRIPTION
**Pull request by ShiftyRail**
[balance]
#22207 killed Bolas by making any shield (at all) having a 50% chance to deflect them, and by making the science powergame armours having 35% chance to deflect them.

This fixes that.

Bolas are a high-risk, high-reward item, compared to tasers, they're faster but have only one charge and can give your opponent a weapon to stun you from a distance if you miss.
The ghetto variant is also a very useful self-defence tool.

I don't believe their main selling point (going through shields and armour) needed to be nerfed.

:cl:
- tweak: Shields, reactive teleport armor, and eswords, will no longer protect you from thrown bolas.

### CODE REVIEW

**Comment by Kharkov72** [comment]
Hit or miss, I guess they never miss huh?

File: code/defines/obj/weapon.dm [hunk 1]
```diff
@@ -197,6 +197,7 @@
 	breakouttime = 50 //10 seconds
 	throw_speed = 1
 	throw_range = 10
+	alwayspass = TRUE // Hit or miss? I guess they never miss
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/defines/obj/weapon.dm]: 😞

**Comment by ghost** [comment]
Why can tazer shots miss and hit shields but bolas can't? Even lasers can be blocked by riot shields.

**Comment by Kurfursten** [comment]
That's a lot of antagonist nerfs from you in such a short period. Ninjas don't need any nerfs like this and frankly reactive teleport armor is almost never used, I don't know why you think it needs nerfing.

**Comment by ShiftyRail** [comment]
Because bolas are much more situational than tasers ? They got to have some advantages.
The stun is ridiculously low, being about 4 seconds, so you have to follow up with something anyways.

@Kurfursten 
I'm sorry but this is now a safe space for our security staff, be less #redphobic

My lad this killed Bolas on a much more general level than just ninja
Code an exception for it if you feel like it.

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -930,6 +930,7 @@
 //Called when the item blocks an attack. Return 1 to stop the hit, return 0 to let the hit go through
 /obj/item/proc/on_block(damage, atom/blocked)
 	if(ismob(loc))
+
```
Comment (Kurfursten) [inline code/game/objects/items.dm]: I think you forgot to code the part where this PR does what you said in the changelog?

Comment (ShiftyRail) [reply code/game/objects/items.dm]: I swear I have it on my VS code. Whatever

File: code/defines/obj/weapon.dm (hunk 1 above)
Comment (SonixApache) [reply code/defines/obj/weapon.dm]: HUH

**Comment by boy2mantwicethefam** [comment]
>High-risk

Bruh moment
For real though it's still a stun at the end of the day, making it guaranteed will make it too strong, especially when it bypasses a gimmick armor's gimmick
Consider the following:
1. I was on the receiving end once
2. Bolas auto-throw on whatever you click on, don't even need to set your intent to throw
3. Corridors are still cramped as hell and the bola is a real deal tile-big projectile
4. The stun is long enough to follow up on it, in addition to slowing people down if they somehow manage to escape the stun until they resist out of it

If anything, stun should be removed outright in exchange of guaranteed impact, then it will have an use to make people slow enough to be far easier targets

**Comment by SonixApache** [comment]
just like lower it by half and see if that works

File: code/modules/clothing/suits/armor.dm [hunk 3]
```diff
@@ -234,6 +234,8 @@
 	return
 
 /obj/item/clothing/suit/armor/reactive/on_block(damage, atom/blocked)
+	if (blocked.alwayspass)
+		return TRUE
```
Comment (boy2mantwicethefam) [inline code/modules/clothing/suits/armor.dm]: Should return 0 to be more in line with the rest
1 (TRUE) - Reactive armor reacted and teleported away
0 (FALSE) - Reactive armor didn't react

**Comment by bathosbathtime** [comment]
Having it be a better option to tasers when dealing with shields and armor instead of a full counter to shields and armor would feel much better to me.  I do love the idea of bolas filling a niche in what it can deal with but having it go through shields completely feels bad.  I wouldn't mind if it did/does go through armor since it would be going around the armor anyway.
I mean this as an agreement to sonix's suggestion.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Realistically speaking, and especially pre-flashbang meta, bolas were THE shield/esword counter for longer than /vg/ existed and people still thought they were useless.

This is a return to the status quo, but presented as a sec buff, so people flip shit.

**Comment by Caboozles** [comment]
I think cutting the stuntime would be fair.

I don’t, however, see the fun in permitting RNG to be king when it comes to deciding if these things land. It’s bad enough with tasers.

**Comment by ShiftyRail** [comment]
Bolas are already, by far, the shortest stun in sec arsenal. Even a peperspray, if done correctly, stuns for longer.

**Comment by Blithering** [comment]
I like bolas, but I like shields more. Maybe remove the armour block chance but keep shield block?

**Comment by d3athrow** [comment]
unwanted pomf opinion:
    Remove bolas stun as is
    Bolas can't be blocked or at least block chance reduced
    Attempting to run with bolas on has a chance to make you trip, otherwise it slows your movespeed to walk or less
    Maybe make it so you can just cut the bolas off instead of having to resist out

**Comment by Caboozles** [comment]
Frankly I’m fine with this change as is, but it probably won’t get through without compromise.

**Comment by ShiftyRail** [comment]
Barely anybody used Bolas before this rework because flashbangs were better in everyway and tasers were better.

The projectile rework removed the only niche of interest bolas had.

If someone has a shield, and I have a taser, I can tase twice. Can't do that with a bola.

I find it extremely interesting that this needs compromise but the original change didn't.

**Comment by ghost** [comment]
The original PR was about general throwing objects respecting existing shield system. This PR is about singling out one item to ignore it completely.

**Comment by ShiftyRail** [comment]
Done in a better way.
- Shields have a "blocking" rating (defined by `IsShield()`)
- Projectiles have a "pass" rating

If the pass rating is higher than the blocking rating, it'll always pass.
Bolas have a pass of 1
Holograms have their own snowflake way of handling blocking and aren't affected.

**Comment by coldcola** [comment]
they're bolas, just walk why is buffing an issue

**Comment by ShiftyRail** [comment]
It's not even a buff, it's putting them back at their original behaviour

**Comment by Kitfox88** [comment]
bolas are fun and a reason to use them is nice

**Comment by Allspice** [comment]
@I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE Standard bolas only had 60% success rate up until Bauds' PR a couple years ago, which would explain why nobody used them. And ghetto only had 30% so that didn't leave a good impression either.

**Comment by ShiftyRail** [comment]
Shields (and e-swords) shouldn't be the be-all, end-all solution to ranged combat. 
There is no real argument against this PR in the comments, at all, since

a) I'm not making 'a large number of "a lot of antagonist nerfs from you in such a short period", if you wish to argue that, then prove it, and don't change other people's comment when they snark at you for being absurd, this is just being silly.
I've even made a LOT more PRs to *buff* antags (namely, vampire) than to "nerf" antags.
And even if it were true, what the heck is that argument ? Must a c*der have a quota of buffs/nerfs for his PRs to be taken seriously ?

You'll surely understand that I didn't take your comment seriously at all, and answered with something equally not serious. Why did you edit it, rather than provide actual evidence of that "large number" of nerfs ?

b) This isn't a ninja nerf, the holo projector is not affected in any way
c) This doesn't make bolas better tasers (???) A single use, 5 second stun, with a chance to fail, is not the same as a 7 use ranged 30 seconds stun. (both of which are outclassed by flashbangs)

This PR aims to bring back the status quo ante bellum and to make bolas a viable alternative to tasers in *some* niche situations. Frankly, I just want a bit more versatility in sec arsenal rather than always using flashbang => taser => cuff or lasers if it's lethals.

**Comment by ghost** [comment]
In combat, when you get hit by a stun you're probably done for. However if you are a vampire or can somehow get up again thanks to chems or enemy incompetence, you have no penalties. If you get hit by bolas in combat, you will not recover from forced walking without standing still for a considerable amount of time. Walking or standing still is a death sentence in combat. Bolas as they are, are the best non-lethal disable available if they connect. They even cripple stun immune targets such as pale body cultists.

You want them to have practically no counter at all besides switching to walk intent. Just because people are uninformed about how strong bolas are and don't use them often, doesn't mean they need such treatment. If you can implement a way to mechanically counter them in a significant manner, i.e. chems(?)/shields/swords/armor/gear then I have no problems with it.

**Comment by Reviire** [comment]
Extremely powerful bola technology allows it to bypass teleportation.

**Comment by ShiftyRail** [comment]
Holograms aren't affected. For the third time.

**Comment by yclatious** [comment]
What the fuck is this busta shit? Are people legit going ham on bolas being even remotely half viable? You can have several ranged travel bolts of stun for 1 single inventory space, you can stock up on several deafening screen wide stun nades, you can even just straight up fill a container full of lube and spray it like its a girls fundraiser carwash, but god forbid bolas?

A thing you have to be ultra close to even remotely use? A one use thing? A thing that stuns but has a chance to not actually work? Something that can horribly backfire on you? Something which takes a spot instead of the before mentioned multiple charged projectile stuns and screen stuns? You wanna keep puntkicking that while letting the rest be, over what, sciencedorks?

If an assistant manages to bola a NukeOps with a shield, they deserve it. If a janitor manages to ghetto bola a esword ebow nerd, they deserve it. If sec is Chad enough to want to use Bolas, they have the bloody right to use that shit without being cucked over by shields.

This isnt even a sec buff, this is just giving bolas what they deserve. A spot in the sun, even if lesser than everything else. Let people use the bloody subpar style option for fucks sake, dont gut it.

**Comment by Blithering** [comment]
@yclatious 
>A thing you have to be ultra close to even remotely use?
 
Bolas travel no slower than taser shots, can be loaded into blunderbuss, pneumatic cannon or probably even a fire extinguisher for hitscan. 
>A one use thing?

Reusable, flashbangs are truly one use
>A thing that stuns but has a chance to not actually work?

The electrode narrowly misses !
>Something that can horribly backfire on you?

Kicking flashbangs back at sec is a high level strat
>Something which takes a spot instead of the before mentioned multiple charged projectile stuns and screen stuns?

pocket sized, can carry a box of 7 for the same inventory slot as a 5 shot taser

Meanwhile they can slow down any high mobility (wiz/vamp jaunt) or hulked out target to a crawl indefinitely, are easily spammable and EMP proof, they don't need to also be the be all end all answer to shield wielders. Maybe if a guy spent the tc/took up the inventory space to bring a shield to a fight, he 'deserves' to block a shot or two before losing the coin flip?

**Comment by yclatious** [comment]
Aight then, we doing this.

>Bolas travel no slower than taser shots, can be loaded into blunderbuss, pneumatic cannon or probably even a fire extinguisher for hitscan.

Correction. Pneumo bolas do not work. Unless severe case of RNG has happened, and  20+ test shots of a stock pressurized large air tank bolas on fresh, naked monkeymen failed within a sub 1% chance, the pneumo does not work. Yes, the message did show they hit him. Yes, they fell by his feet. Yes, he was hit by them. No, he never tripped or got bolacuffed. Do not work. Test serb at current date, doubt code is outdated or malfunctioning in any way.

Equaly, fire extinguisher does not work. Dont say probably, back it up with experience.
>The bolas won't fit into the nozzle!

Blunderbuss does work, but such is a highly gimmicky, one shot and then reload ghetto gun which sec is very, very unlikely to get in any way shape or form. And as to the point of being no slower than the taser shots, Id like to argue that leading shots and being able to have several projectiles rushing across the screen, especialy with multiple sec, is much, much more effective than a one and done attempt which grants whoever is fighting you the chance to also stun and legcuff you.

>Reusable, flashbangs are truly one use

And unlike bolas, you get no realistic risk of backfire unless they kick em back. Which I admit, is very stylish but a well cooked flashbang (which any good sec with a stable conection can do) is really something hard to counter, unless you do some matrix tier shit. Bolas always carry risks, which is why I always liked them. Balsy and fun.

>The electrode narrowly misses !

Ill grant you that, one of the few weaknesses of the taser, very fair. Which is exactly my point. 

Let us use other tools for the jobs the taser cannot do, knowing how very niche the proper usage for bolas is amongst security officers. Encourage balsy plays with risk reward tools. Let the underdog gimmick shine atleast in this one field.

>Kicking flashbangs back at sec is a high level strat

As said above, very stylish and I dont disagree. But usualy relegated to bad cooking part on sec. Still a great laugh to do.

>pocket sized, can carry a box of 7 for the same inventory slot as a 5 shot taser

I could see your point, if you really didnt also have flashbangs in your pocket, which almost guarantees arest if used decently, a taser in your armor slot with ranged stuns you can lead and an absolute killer in a tight hallway/maint if you dont just blunder it, so on. Every bola you miss is another tool for an antag to use agaisnt you and others. 

A stun you land as sec will almost always end with the perp captured or dead, with its duration and general circumstances. Being able to carry 7, as of now, weak agaisnt shields mostly gimmick items, which very much do go agaisnt you if you miss, is nothing more outrageous than getting a box full of flashies. And you know for a fact flashies will get you more results, even agaisnt shielded dudes.

>Meanwhile they can slow down any high mobility (wiz/vamp jaunt) or hulked out target to a crawl indefinitely, are easily spammable and EMP proof, they don't need to also be the be all end all answer to shield wielders. Maybe if a guy spent the tc/took up the inventory space to bring a shield to a fight, he 'deserves' to block a shot or two before losing the coin flip?

Wizard and Vamp jaunts being slowed by bolas sounds somewhat silly honestly, but I guess its a feature? About hulks being slowed, Id call it a good mechanic to actually slow em down. Why is the fact a tool that is generaly not used being used in niche cases a bad thing? Why is balsy play discouraged?

Why can a counter to shields not exist? They arent the be all end all answer to shields, that still mostly remains as bangers, since they counter almost everything, really. Why must a TC investment be considered sacred in any way, when there are always counters to everything that a dude with some time can just do? A single lube extinguisher can delete a Death Squid, a well time flashbang can utterly cuck an entire nuke ops batallion, a well placed prox welder bomb can utter ruin someone and even get a shuttle called, so why cant a single madman sec/greyshit with a dream in his heart and a bola in his hand not be able to deal with one person?

Nobody -deserves- anything. Shields dont have guaranteed taser bolt denial, its rng. Eswords, same shit. You can wear double eshields and youl still get hit, heck, in the small test I did, the first 3 stuns, with double shields, were in, by order, 2 bolts, 2 bolts, 1 bolt.The following one took 10. You cant seriously argue for fairness agaisnt shields when even the turbo defensive option can, and has gotten dunked with just 1 or 2 taser shots. There is no guaranteed defense agaisnt a certain number, its all just random.

There is no -deserves- by the code itself and the way it works. This "buff", which is just a normalization, deserves to exist and there isnt any real reason to deny it. Sec has plenty of tools to deal with almost anything you throw at it, eefficiency depending on skill level, but to have a tool both greytide and security can use, for a very specific situation, IN WHICH
-IT IS NOT INHERENTLY MUCH BETTER 
-IS RISKIER
-HAS MORE CONSEQUENCES IF YOU MISS, GIVING THE ATACKER A VALID WEAPON AGAISNT YOU

Give bolas the niche they rightfully deserve, and quit justifying on bad examples that are non valid, speculation and really, stuff thats more fluff than anything. If you wanted serious fairness for eshields, youd code an actual determined stun absorbtion number instead of just rng.

Thank you for coming to my TED talk.

**Comment by Blithering** [comment]
time to spam more walls of text, but I'll try to keep this brief. Don't want to turn this pr discussion into a novel any more than it is already.
Bolas have a niche, that's the legcuff on hit, no other projectile does this. They don't need a second one just because people can throw them back if you miss.
Shields are at present, really luck based, and batons already bypass them entirely. More ways to bypass them will just make them more crap.
To clarify, bolas don't slow down people while jaunting, but they mean that when they come out of the jaunt they either have to stand still for thirty seconds or limp away at bay speed. this makes them incredibly easy to track.
I don't know why you're calling this buff a normalisation, the nerf was the normalisation, making thrown items respect armour and shields like other projectiles or attacks do. 
TLDR: Bolas are good enough already, shields are crap but will be even worse if this gets merged.

**Comment by yclatious** [comment]
Nothing needs more than a second niche, some do, some dont. Bolas being good is a non issue, and like I said above, -deserve- doesnt exist in the code.
No, if you want shields to be good, make them actually good instead of RNG, but ungunking bolas is a good thing for now and permits you to tinker with balance in the future.
Bolas dont take thirty seconds, to my knowledge. A vamp/wiz jaunting into a corner of maint is not easily tracked if he knows where he is going, especialy in places like box and deff.
It is a normalization because to me, they were always the shield ignoring tool. They were always, for the most part, weak, and I recognised this small thing as their one role, their time to shine. Consider it neglection feature spawning syndrome.
Bolas have always been inferior. For the love of whatever you like, atleast allow them to have one good damn trait and to be anything but trash tier for most and used by goddamn greyshits, heck, dont nerf greyshits too.
Bolas are not good enough, shields need work outside of this, they will not be worse because lets also face it, shits so rng and few men are mad enough to be balsy and risk it that the hypothetical of a "shield meta" would never be broken. This gutted an item that always was gutted and finaly became okay. 

Stop gutting fun. Encourage balsyness. Merge the PR.

**Comment by Blithering** [comment]
You're contradicting yourself again, first you say that anyone who throws a bolas at a shield wielding guy -deserves- to have a guaranteed trip, then you say there's no such thing as -deserve-, now you're saying that you want to encourage ballsy moves by by making the thing that 'few men are mad enough to be ballsy and risk' have a guaranteed ranged counter. Double shield is fun, a stun for every situations isn't.

**Comment by yclatious** [comment]
>You're contradicting yourself again, first you say that anyone who throws a bolas at a shield wielding guy -deserves- to have a guaranteed trip, then you say there's no such thing as -deserve-, now you're saying that you want to encourage ballsy moves by by making the thing that 'few men are mad enough to be ballsy and risk' have a guaranteed ranged counter.

And yet Im not saying they deserve it, because bolas can and still do fail, even when they should hit. Im saying the code never gave anyone anything they -deserve- because its all RNG but Id like for some of the RNG to be stripped away in order to encourage balsy play. It isnt making a guaranteed ranged counter, because that still remains the lube extinguisher and the flashbang for area denial, stuns and unblockable slips. A bola, a thing you can very much miss, can be used agaisnt you, yadayada, also being decent is not a crime in any way, shape or form.

>Double shield is fun, a stun for every situations isn't.

The thing is, double shield isnt really fun. Its still governed by absolute luck. Like I said above, 1 taser bolt or 2 can still delete it, sometimes its 10, sometimes it doesnt block anything, and lets also be real, nobody is going to use both and be a walking looty pinato for the sake of something that rarely exists. Even with double protective values, its finicky at best, gimmicky at its core, useless in almost every single one of its execution attempts, so why should it only specificaly ruin the one, mark it, the one balsy as hell way to play chad sec? A

Also you drop both shields on stun, so its even worse, cant even keep them on your hands.

There have been stuns for every single situation and no code to nerf everything, so why the heck are people dogpiling on bolas, the absolute underdogs with limited as heck range and niche execution. It objectively makes no sense, especialy when people go ahead and theory craft about pneumos and extinguishers without actually checking if they even work.

Its scaremongering over the chad mode of arest, really. I get the aspect of "stop buffing sec" but lets also be sincere lads, this buffs greytide the most. And I know for a fact I want to cheer the greyshit with a dream in his heart and bolas on his hands that goes toe to toe with the deathsquids and wins with nothing but a ian and runtime bola.

**Comment by Blithering** [comment]
>I know for a fact I want to cheer the greyshit with a dream in his heart and bolas on his hands

Cheering for that assistant is exactly why this PR shouldn't pass. You feel for a guy going against the odds to achieve the improbable, but when you remove the likelihood of failure, it stops being something to root for, and just becomes a guy picking paper because he see's a rock coming. That's why dual shields is the meme dream, that one time you stand your ground and block a hail of taser bolts is all the sweeter knowing that any one of those could have landed you on your ass and got you killed.

**Comment by yclatious** [comment]
>You feel for a guy going against the odds to achieve the improbable, but when you remove the likelihood of failure, it stops being something to root for, and just becomes a guy picking paper because he see's a rock coming.

Except often its a greyshit with ghetto bolas, which already have a around 60% chance to work, -without shields-, IF they hit (improbable), can already lose components of the bola upon failure, making them useless, impossible to repair in the heat of the battle and, if we go by my example, then boyo, you have maybe, what, two bolas with low as fuck chances (not even counting shields into it) agaisnt lads with several mags worth of on hit stun damage with can and will lock you till you are swiss cheese, not even mentioning the other tools they have.

Even before this? It was already a fucking mad lad move. With this? Its not even a mad lad move, its utter trash.

To prove this point, I decided to be a thickheaded idiot and do some testing, some very extensive, extremely boring testing.

With ghetto bolas, used on a nukiesquad armor wearing monkeymen, one e-shield, it took, from the 5 tests I have done in the test server:
-7 throws, 0 breaks, trip, probabilistic anomaly.
-101 throws, 10 bola desintegrations, no trip, decided to skip as results were beyond telling.
-60 throws, 10 desintegrationsn, no trip, for sake of quickness, have reduced to 30 sample size.
-30 throws, 3 desintegrations, no trip.
-Also 30 throws, 3 desintegrations, no trip.
Name under which it was done was Joshua Best, you can check the test server logs, Im sure you will find Im not kidding.

101+7+60+30+30=108+120=138 throws. One singular trip attempt successfull amongst 138 throws. 

Either I am RNG cursed, or this is practicaly impossible to execute in a normal game, by any metric you imagine or employ, by simple usage of 1 singular eshield. And yes, I did check if they were legcuffed every so often. Nothing.

And yes, **I did do 138 throws and rebuilt the bola 26 times just to make a point.** Bolas are, as of now, GUNKED. They deserve better than this shit agaisnt fucking shields. One hundred and thirty fucking eight throws and ONE SINGULAR TRIP is utterly unnaceptable for a ghetto bola.

This is not rootable for, this is suicide. This pr is not rock to paper, this is making it usable. Double shields can already drop from 1 singular taser bolt, as I have tested before, so no, a hallway of tasers wont be blocked by double shields. I have objectively just proven they are beyond weak, they are unusable in any go hard greyshit saving the station situation. They are utterly fucked.

Removing the likelyhood of failure? This removed the likelyhood of success. Merge the goddamn PR.

**Comment by Kurfursten** [comment]
In light of the design change to make this not affect ninja balance, and the recent buff to reactive teleport armor that just went through, this seems acceptable, particularly with the public support for it.
